tests/integration/aps/test_fetch_from_api.py

```diff
@@ -0,0 +1,109 @@
+import datetime
+
+import pytest
+from airflow import DAG, AirflowException
+from airflow.models import DagBag
+from airflow.models.dagrun import DagRun
+from airflow.utils.state import DagRunState
+from aps.APSApiClient import APSApiClient
+from aps.APSParams import APSParams
+from aps.fetch_from_api import save_file_in_s3, set_APS_harvesting_interval
+from aps.repository import APSRepository
+from busypie import SECOND, wait
+
+DAG_NAME = "aps_fetch_api"
+TRIGGERED_DAG_NAME = "aps_process_file"
+
+
+@pytest.fixture
+def dag():
+    dagbag = DagBag(dag_folder="dags/", include_examples=False)
+    assert dagbag.import_errors.get(f"dags/{DAG_NAME}.py") is None
+    return dagbag.get_dag(dag_id=DAG_NAME)
+
+
+def test_dag_loaded(dag: DAG):
+    assert dag is not None
+    assert len(dag.tasks) == 3
+
+
+def test_dag_trigger_file_processing(dag: DAG):
+    repo = APSRepository()
+    repo.delete_all()
+    assert len(repo.find_all()) == 0
+
+    id = datetime.datetime.utcnow().strftime("test_aps_fetch_api_%Y-%m-%dT%H:%M:%S.%f")
+    dagrun = dag.create_dagrun(DagRunState.QUEUED, run_id=id)
+    wait().at_most(60, SECOND).until(lambda: __test_dagrun_state(dagrun))
+
+    queued_tasks = get_dag_runs_amount(TRIGGERED_DAG_NAME, state="queued")
+    running_tasks = get_dag_runs_amount(TRIGGERED_DAG_NAME, state="running")
+    queued_and_running_tasks = queued_tasks + running_tasks
+    if len(repo.find_all()) == 0:
+        assert 0 == queued_and_running_tasks
+    else:
+        assert queued_and_running_tasks > 0
```

Comment:
I don't understand this test as it is not deterministic. One of the two cases should be true, why is it possible to have both cases?

Reply:
yes, it can be both, because if there was no articles uploaded today, no json will be saved on S3

Reply:
it's possible because is not on us: if there was no articles uploaded to day, no json will be downloaded